### PR TITLE
Use minimal SDK

### PR DESCRIPTION
**Pull request by jonesbusy**
Currently migrating to external storage (artifact, etc..) and want to migrate job caches to S3 endpoint

Similar to S3 publisher : https://plugins.jenkins.io/s3/dependencies/, no need to bring all AWS dependencies (https://plugins.jenkins.io/aws-java-sdk-minimal/).

I'm not sure if there is a better migration path expect asking consumer to install the minimal SDK.

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by repolevedavaj** [comment]
Sorry for not yet replying. I will check it in the next days.

**Comment by jonesbusy** [comment]
@repolevedavaj Hi any update there ?

**Comment by repolevedavaj** [comment]
@jonesbusy Yes i checked but wanted to contact you: What is the benefit of using the minimal plugin?

**Comment by jonesbusy** [comment]
@repolevedavaj The size. The minimal was created to optimize the size. It only contains core and basic service like S3. All the rest is not used by this plugin

The difference is insane

3.68 MB for the minimal https://repo.jenkins-ci.org/artifactory/releases/org/jenkins-ci/plugins/aws-java-sdk/aws-java-sdk-minimal/1.12.586-413.v6a_6c3a_420126/ 
273.97 MB for the all https://repo.jenkins-ci.org/artifactory/releases/org/jenkins-ci/plugins/aws-java-sdk/1.12.586-413.v6a_6c3a_420126/

If you want to keep the all flavor (But I don't want to see a valid reason) can you perform the conditional check on the descriptor ?

From https://www.jenkins.io/doc/developer/plugin-development/optional-dependencies/

```java
if (Jenkins.getInstance().getPlugin("aws-java-sdk") != null || Jenkins.getInstance().getPlugin("aws-java-sdk-minimal") != null ) {
    
}
```

It don't bring any benefits in my opinion. Having user installing the minimal one even if they have already the `aws-java-sdk` is not a big deal. But forcing user to add 200MB of size per instance is just a no go.

Thanks!

File: pom.xml [hunk 1]
```diff
@@ -59,9 +58,8 @@
             <artifactId>variant</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>aws-java-sdk</artifactId>
-            <version>${aws-java-sdk.version}</version>
+            <groupId>org.jenkins-ci.plugins.aws-java-sdk</groupId>
+            <artifactId>aws-java-sdk-minimal</artifactId>
```
Comment (jonesbusy) [inline pom.xml:62]: Part of bom

**Comment by repolevedavaj** [comment]
@jonesbusy OK I see your point and I agree that trying to maintain backwards compatibility in this case is probably not that useful.

**Review by repolevedavaj — APPROVED** [review]

**Pull request merged**